recipes/ray-packages/meta.yaml

```diff
@@ -0,0 +1,256 @@
+  number: 0
+  skip: True  # [py<36]
+  # skip on MacOS as there's some weird compilation issue and I have no MacOS to develop on
+  skip: True  # [osx]
+
+# Need these up here for conda-smithy to handle them properly.
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cython >=0.29
+    - bazel <=3.4.1  # [not win]
+    - bazel  # [win]
+    - curl
+    - make
+    - m2-patch  # [win]
+
+outputs:
+  - name: ray-all
+    requirements:
+      host:
+      run:
+        - python
+        - {{ pin_subpackage('ray-core', exact=True) }}
+        #- {{ pin_subpackage('ray-debug', exact=True) }} # temp turned off, see below
+        - {{ pin_subpackage('ray-dashboard', exact=True) }}
+        # some rllib deps are missing on Windows
+        - {{ pin_subpackage('ray-rllib', exact=True) }}  # [not win]
+        - {{ pin_subpackage('ray-serve', exact=True) }}
+        - {{ pin_subpackage('ray-tune', exact=True) }}
+        # autoscaler does not work on Windows, no point of building it
+        - {{ pin_subpackage('ray-autoscaler', exact=True) }}  # [not win]
+    test:
+      imports:
+        # dummy test; actual tests are in subpackages
+        - ray
+
+  - name: ray-core
+    script: build-core.sh  # [not win]
+    script: build-core.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cython >=0.29
+        - bazel <=3.4.1  # [not win]
+        - bazel  # [win]
+        - curl
+        - make
+      host:
+        # TODO: check which of these can move to run-reqs
+        - click
+        - cloudpickle
+        - colorama
+        - filelock
+        - funcsigs
+        - jsonschema
+        - numpy >=1.16
+        - pickle5  # [py<38]
+        - packaging
+        - protobuf >=3.8.0
+        - pytest
+        - pyyaml
+        - redis-py >=3.3.2, <3.5.0
+        - python
+        - pip
+      run:
+        - python
+        - aiohttp
+        - aiohttp-cors
+        - aioredis
+        - click >=7.0
+        - colorama
+        - colorful
+        - filelock
+        - googlesearch
+        # gpustat-0.6.0 has a dependency which does not exist on Windows;
+        # skip it there until gpustat is fixed as it is optional
+        - gpustat  # [not win]
+        - grpcio >=1.28.1
+        - jsonschema
+        - msgpack-python >=1.0.0, <2.0.0
+        - numpy >=1.16
+        - protobuf >=3.8.0
+        - psutil
+        - pyyaml
+        - requests
+        - redis-py >=3.3.2, <3.5.0
+        - opencensus
+        - prometheus_client >=0.7.1
+        - pickle5  # [py<38]
+        - funcsigs
+        - setproctitle
+    test:
+      imports:
+        - ray
+        - ray._raylet
+        - ray.actor
+        - ray.profiling
+        - ray.runtime_context
+        - ray.state
+        - ray.worker
+
+  - name: ray-debug
+    build:
+      # some weird incompatibility when installing during tests... disabling for now
+      skip: True  # [not win]
```

Comment:
Can you provide a bit more detail on this? Or can you retest it in CI and post some results?

Reply:
As far as I was able to figure out, `py-spy` uses newer version of `glibc` and `sysroot`, which causes quite weird issues during environment creation - `conda` refuses to create an env with it (but happily installs `py-spy` in an env created without it).

That said, I've seen this a few weeks ago, so something might have changed.

Also, I've looked at the subpackage now again, and I see that all it does it installs `py-spy`, which seems a little bit pointless. Maybe we should just drop this subpackage altogether?

Reply:
I have now faced very similar issue with other subpackage on Windows, where it fails creating the environment during tests in conda-build, but happilly creates the environment later on.

I now start to believe this is some strange metadata caching which _might_ be going away if I bump build number...

Reply:
Okay, here's the relevant error - I've enabled `py-spy` and `ray-debug`.
Here's the mayhem: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=248890&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=13008

Reply:
Before the azure logs get deleted, here's the pertinent part:

```
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.

Building graph of deps:   0%|          | 0/8 [00:00<?, ?it/s]
Examining python[version='>=3.6,<3.7.0a0']:   0%|          | 0/8 [00:00<?, ?it/s]
Examining py-spy[version='>=0.2.0']:  12%|█▎        | 1/8 [00:00<00:02,  3.45it/s]
Examining py-spy[version='>=0.2.0']:  25%|██▌       | 2/8 [00:00<00:00,  6.89it/s]
Examining ray-debug==1.0.1=py36h5fab9bb_0:  25%|██▌       | 2/8 [00:00<00:00,  6.89it/s]
Examining @/linux-64::__glibc==2.12=0:  38%|███▊      | 3/8 [00:00<00:00,  6.89it/s]    
Examining @/linux-64::__archspec==1=x86_64:  50%|█████     | 4/8 [00:00<00:00,  6.89it/s]
Examining ray-core==1.0.1=py36hdf6fd44_0:  62%|██████▎   | 5/8 [00:00<00:00,  6.89it/s]  
Examining python_abi=3.6[build=*_cp36m]:  75%|███████▌  | 6/8 [00:00<00:00,  6.89it/s] 
Examining python_abi=3.6[build=*_cp36m]:  88%|████████▊ | 7/8 [00:00<00:00,  9.25it/s]
Examining @/linux-64::__unix==0=0:  88%|████████▊ | 7/8 [00:00<00:00,  9.25it/s]      
                                                                                

Determining conflicts:   0%|          | 0/8 [00:00<?, ?it/s]
Examining conflict for ray-core python_abi python ray-debug:   0%|          | 0/8 [00:00<?, ?it/s]
Examining conflict for python ray-debug py-spy ray-core python_abi:  12%|█▎        | 1/8 [00:00<00:01,  5.48it/s]
Examining conflict for python ray-debug py-spy ray-core python_abi:  25%|██▌       | 2/8 [00:00<00:00, 10.95it/s]
Examining conflict for ray-core python py-spy:  25%|██▌       | 2/8 [00:00<00:00, 10.95it/s]                     
Examining conflict for ray-core python py-spy:  38%|███▊      | 3/8 [00:00<00:00,  6.46it/s]
Examining conflict for ray-debug py-spy:  38%|███▊      | 3/8 [00:00<00:00,  6.46it/s]      
Examining conflict for ray-debug py-spy:  50%|█████     | 4/8 [00:00<00:00,  4.49it/s]
Examining conflict for __glibc ray-debug py-spy:  50%|█████     | 4/8 [00:00<00:00,  4.49it/s]
Examining conflict for ray-core ray-debug:  62%|██████▎   | 5/8 [00:00<00:00,  4.49it/s]      
                                                                                        
failed
failed to get install actions, retrying.  exception was: Unsatisfiable dependencies for platform linux-64: {'feature:|@/linux-64::__glibc==2.12=0', "_libgcc_mutex[version='*|0.1',build='main|conda_forge']", '_libgcc_mutex==0.1=conda_forge', "libgcc-ng[version='>=4.9|>=7.3.0']", 'ray-core==1.0.1=py36hdf6fd44_0', "__glibc[version='>=2.17,<3.0.a0']", 'python_abi=3.6[build=*_cp36m]', "python_abi[version='2.7.*|3.6|3.7.*|3.9.*|3.8.*',build='*_cp27mu|*_cp38|*_cp37m|*_cp39|*_pypy36_pp73']", "libgcc-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=9.3.0|>=7.2.0']", "libgcc-ng[version='>=9.3.0']", "libgcc-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=7.2.0']", 'feature:/linux-64::__glibc==2.12=0', "libgcc-ng[version='>=7.5.0']", 'python_abi==3.6[build=*_pypy36_pp73]'}
WARNING conda_build.build:test(2799): failed to get install actions, retrying.  exception was: Unsatisfiable dependencies for platform linux-64: {'feature:|@/linux-64::__glibc==2.12=0', "_libgcc_mutex[version='*|0.1',build='main|conda_forge']", '_libgcc_mutex==0.1=conda_forge', "libgcc-ng[version='>=4.9|>=7.3.0']", 'ray-core==1.0.1=py36hdf6fd44_0', "__glibc[version='>=2.17,<3.0.a0']", 'python_abi=3.6[build=*_cp36m]', "python_abi[version='2.7.*|3.6|3.7.*|3.9.*|3.8.*',build='*_cp27mu|*_cp38|*_cp37m|*_cp39|*_pypy36_pp73']", "libgcc-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=9.3.0|>=7.2.0']", "libgcc-ng[version='>=9.3.0']", "libgcc-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=7.2.0']", 'feature:/linux-64::__glibc==2.12=0', "libgcc-ng[version='>=7.5.0']", 'python_abi==3.6[build=*_pypy36_pp73]'}
```

Something appears to be depending on a very old glibc (`feature:|@/linux-64::__glibc==2.12=0`), so I would concur with your guess... Not that important now IMO.

Reply:
I think this very old glibc is the one that is present in Docker image which conda-forge CI uses...